crypto/src/secp256k1.rs

```diff
@@ -231,12 +231,20 @@ impl<'de> Deserialize<'de> for Secp256k1Signature {
 
 impl Signature for Secp256k1Signature {
     fn from_bytes(bytes: &[u8]) -> Result<Self, signature::Error> {
-        let recovery_id = rust_secp256k1::ecdsa::RecoveryId::from_i32(bytes[64] as i32).unwrap();
-        match rust_secp256k1::ecdsa::RecoverableSignature::from_compact(&bytes[..64], recovery_id) {
-            Ok(sig) => Ok(Secp256k1Signature {
-                sig,
-                bytes: OnceCell::new(),
-            }),
+        if bytes.len() != 65 {
+            return Err(signature::Error::new());
+        }
+        match rust_secp256k1::ecdsa::RecoveryId::from_i32(bytes[64] as i32) {
+            Ok(rec_id) => match rust_secp256k1::ecdsa::RecoverableSignature::from_compact(
+                &bytes[..64],
+                rec_id,
+            ) {
+                Ok(sig) => Ok(Secp256k1Signature {
+                    sig,
+                    bytes: OnceCell::new(),
+                }),
+                Err(_) => Err(signature::Error::new()),
+            },
```

Comment:
```suggestion
RecoveryId::from_i32(bytes[64] as i32)
            .and_then(|rec_id| {
                RecoverableSignature::from_compact(&bytes[..64], rec_id).map(|sig| {
                    Secp256k1Signature {
                        sig,
                        bytes: OnceCell::new(),
                    }
                })
            })
            .map_err(signature::Error::from_source)
```

Reply:
(thanks for bearing with my rusty rust, looks so much better 💯 )

```
**NOTE:** The "source" should **NOT** be used to propagate cryptographic errors e.g. signature parsing or verification errors. 
```
as we are parsing signature here, this may not be suitable